src/sentry/integrations/vercel/integration.py

```diff
@@ -248,55 +248,57 @@ def update_organization_config(self, data):
             )
 
             is_next_js = vercel_project.get("framework") == "nextjs"
-            dsn_secret_name = "NEXT_PUBLIC_SENTRY_DSN_%s" if is_next_js else "SENTRY_DSN_%s"
-
-            secret_names = [
-                "SENTRY_ORG_%s" % uuid,
-                "SENTRY_PROJECT_%s" % uuid,
-                dsn_secret_name % uuid,
-                "SENTRY_AUTH_TOKEN_%s" % uuid,
-            ]
-            values = [
-                sentry_project.organization.slug,
-                sentry_project.slug,
-                sentry_project_dsn,
-                sentry_auth_token,
-            ]
-
             dsn_env_name = "NEXT_PUBLIC_SENTRY_DSN" if is_next_js else "SENTRY_DSN"
 
-            env_var_names = [
-                "SENTRY_ORG",
-                "SENTRY_PROJECT",
-                dsn_env_name,
-                "SENTRY_AUTH_TOKEN",
-                "VERCEL_%s_COMMIT_SHA" % source_code_provider.upper(),
-            ]
-
-            secrets = []
-            for name, val in zip(secret_names, values):
-                secrets.append(vercel_client.create_secret(vercel_project_id, name, val))
+            env_var_map = {
+                "SENTRY_ORG": {"type": "plain", "value": sentry_project.organization.slug},
+                "SENTRY_PROJECT": {"type": "plain", "value": sentry_project.slug},
+                dsn_env_name: {"type": "plain", "value": sentry_project_dsn},
+                "SENTRY_AUTH_TOKEN": {"type": "secret", "value": sentry_auth_token},
+                "VERCEL_GIT_COMMIT_SHA": {"type": "system", "value": "VERCEL_GIT_COMMIT_SHA"},
+            }
 
-            secrets.append("")
-            for secret, env_var in zip(secrets, env_var_names):
-                self.create_env_var(vercel_client, vercel_project_id, env_var, secret)
+            for env_var, details in env_var_map.items():
+                if details["type"] == "secret":
+                    secret_name = env_var + "_%s" % uuid
+                    secret_value = vercel_client.create_secret(secret_name, details["value"])
+                    self.create_env_var(
+                        vercel_client, vercel_project_id, env_var, secret_value, details["type"]
+                    )
+                else:
+                    self.create_env_var(
```

Comment:
@ceorourke Nit: we can combine this into just one call of `create_env_var` if you set `details["value"]` to `secret_value`